### DISCUSSION TITLE
Query about form getInputProps

### DISCUSSION THREAD

**Discussion by DJWalker42** [mantinedev/mantine#4745] [Q&A] [answered]
Mantine v6.x.x, react v.18.x.x

Should `getInputProps` be creating a new form path/field when it is passed a path that does not exist in the `initialValues` passed to the `useForm` hook?

To illustrate I've done a brief example in [codesandbox](https://codesandbox.io/s/mantine-playground-forked-yv9j6m?file=/src/App.tsx) with the following code:
```
import { MantineProvider, TextInput} from '@mantine/core';
import { useForm } from '@mantine/form';

export default function App() {
  const form = useForm({
    initialValues: {
      a: '1',
      b: '2',
      c: '3'
    },
  });

  return (
    <MantineProvider>
      <form>
        <TextInput {...form.getInputProps('z')}></TextInput>
      </form>  
      <p>{JSON.stringify(form.values)}</p>
    </MantineProvider>
  );
}
```

When the `TextInput` box has something inputted the form now has a new field "z". Is this the expected behaviour?

**Comment by rtivital** [comment, marked as answer]
use-form hook is designed to work with data schemas with known shape. Although it will work in your example, it is not recommended to use it this way as it will break in more complex scenarios (like nested fields). Moreover, the input will be switched from uncontrolled to controlled when the first character is entered, and you will get a warning in the console from React.
